ui/page/dexclient/create_wallet_modal.go

```diff
@@ -0,0 +1,253 @@
+	md := &createWalletModal{
+		Load:             l,
+		modal:            l.Theme.ModalFloatTitle(),
+		walletPassword:   l.Theme.EditorPassword(new(widget.Editor), "Wallet Password"),
+		appPassword:      l.Theme.EditorPassword(new(widget.Editor), "App Password"),
+		createNewWallet:  l.Theme.Button("Add"),
+		walletInfoWidget: wallInfo,
+		walletName:       l.Theme.Editor(new(widget.Editor), "Wallet Name"),
+		RPCAddress:       l.Theme.Editor(new(widget.Editor), "RPC Address"),
+		RPCPort:          l.Theme.Editor(new(widget.Editor), "RPC Port"),
+		RPCUsername:      l.Theme.Editor(new(widget.Editor), "RPC username"),
+		RPCPassword:      l.Theme.EditorPassword(new(widget.Editor), "RPC Password"),
+	}
+
+	md.createNewWallet.TextSize = values.TextSize12
+	md.createNewWallet.Background = l.Theme.Color.Primary
+	md.appPassword.Editor.SingleLine = true
+	md.appPassword.Editor.SetText("")
+
+	md.sourceAccountSelector = components.NewAccountSelector(md.Load).
+		Title("Sellect DCR account to pay fee").
+		AccountSelected(func(selectedAccount *dcrlibwallet.Account) {
+
+		}).
+		AccountValidator(func(account *dcrlibwallet.Account) bool {
+			// Filter out imported account and mixed.
+			wal := md.WL.MultiWallet.WalletWithID(account.WalletID)
+			if account.Number == load.MaxInt32 ||
+				account.Number == wal.MixedAccountNumber() {
+				return false
+			}
+			return true
+		})
+	err := md.sourceAccountSelector.SelectFirstWalletValidAccount()
+	if err != nil {
+		md.Toast.NotifyError(err.Error())
+	}
+
+	return md
+}
+
+func (md *createWalletModal) ModalID() string {
+	return dexCreateWalletModalID
+}
+
+func (md *createWalletModal) Show() {
+	md.ShowModal(md)
+}
+
+func (md *createWalletModal) Dismiss() {
+	md.DismissModal(md)
+}
+
+func (md *createWalletModal) OnDismiss() {
+}
+
+func (md *createWalletModal) OnResume() {
+}
+
+func (md *createWalletModal) Handle() {
+	if md.createNewWallet.Button.Clicked() {
+		if strings.Trim(md.appPassword.Editor.Text(), " ") == "" || md.isSending {
+			return
+		}
+
+		md.isSending = true
+		go func() {
+			defer func() {
+				md.isSending = false
+			}()
+
+			coinID := md.walletInfoWidget.coinID
+			coinName := md.walletInfoWidget.coinName
+			has := md.Dexc.WalletState(coinID) != nil
+			if has {
+				md.Toast.NotifyError(fmt.Sprintf("already connected a %s wallet", coinName))
+				return
+			}
+
+			var selectedDcrWallet *dcrlibwallet.Wallet
+			if coinID == dcr.BipID {
+				selectedDcrWallet = md.Load.WL.MultiWallet.WalletWithID(md.sourceAccountSelector.SelectedAccount().WalletID)
+				md.Dexc.SetWalletForDcrAsset(selectedDcrWallet)
+			}
+
+			settings := make(map[string]string)
+			var walletType string
+			appPass := []byte(md.appPassword.Editor.Text())
+			walletPass := []byte(md.walletPassword.Editor.Text())
+
+			switch coinID {
+			case dcr.BipID:
+				settings["account"] = md.sourceAccountSelector.SelectedAccount().Name
```

Comment:
Looks like we can do away with that if statement above and set the dcrwallet here:

```suggestion
			case dcr.BipID:
				selectedAccount := md.sourceAccountSelector.SelectedAccount()
				selectedDcrWallet = md.Load.WL.MultiWallet.WalletWithID(selectedAccount.WalletID)
				md.Dexc.SetWalletForDcrAsset(selectedDcrWallet)
				settings["account"] = selectedAccount.Name
```